tests/hpc/slurm_master_db.pm

```diff
@@ -18,6 +18,165 @@ use warnings;
 use testapi;
 use lockapi;
 use utils;
+use Data::Dumper;
+use Mojo::JSON;
+
+## TODO: provide better parser for HPC specific tests
+sub get_test_name {
+    return my $name = (caller(2))[3];
+}
+
+sub validate_result {
+    my ($result) = @_;
+
+    if ($result == 0) {
+        return 'PASS';
+    } elsif ($result == 1) {
+        return 'FAIL';
+    } else {
+        return undef;
+    }
+}
+
+sub results {
+    my ($description, $result) = @_;
+
+    my %results = (
+        test        => get_test_name(),
+        description => $description,
+        result      => validate_result($result)
+    );
+    return %results;
+}
+
+sub pars_results {
+    my (%test)  = @_;
+    my $file    = 'tmpresults.xml';
+    my @results = keys %test;
+
+    my $test_name;
+    my $description;
+    my $result;
+
+    for my $i (@results) {
+        if ($i eq 'test') {
+            $test_name = $test{$i};
+        } elsif ($i eq 'description') {
+            $description = $test{$i};
+        } elsif ($i eq 'result') {
+            if ($test{$i} eq 'FAIL') {
+                $result = $test{$i};
+            }
+        } else {
+            die('Wrong key!');
+        }
+    }
+
+    if ($result eq 'FAIL') {
+        script_run("echo \"<testcase name='$test_name' errors='1'>\" | tee -a $file");
+    } else {
+        script_run("echo \"<testcase name='$test_name'>\" | tee -a $file");
+    }
+    script_run("echo \"<system-out>\" | tee -a $file");
+    script_run("echo $description | tee -a $file");
+    script_run("echo \"</system-out>\" | tee -a $file");
+    script_run("echo \"</testcase>\" | tee -a $file");
+}
+
+sub test_01 {
```

Comment:
I would recommend you to come up with sane tests naming . for me it sounds really strange when someone saying "I am working on test 42 for today  ". if you want to keep some order take a look into approach taking by wicked tests

Reply:
Yup, I was of course thinking about it. First of all I added $name as Clemens suggested in the %hash holding name, description and results.

But overall I plan on 'grouping' the test in the manner you suggest. But I'm not sure yet which way would be good ;-) so I will keep this improvement for late. OK?

Reply:
fine for me